pkg/appmanager/outputConfig.go

```diff
@@ -61,26 +119,95 @@ func (appMgr *Manager) outputConfigLocked() {
 	// it must have at least one client or server SSL profile associated with
 	// it. If the virtual doesn't have any of either type, we force it to take
 	// the BIG-IP's base client SSL profile in the output config.
-	for vKey, virtual := range resources.Virtuals {
-		for _, irule := range virtual.IRules {
-			if strings.Contains(irule, sslPassthroughIRuleName) {
-				clientProfCt := virtual.GetProfileCountByContext(customProfileClient)
-				serverProfCt := virtual.GetProfileCountByContext(customProfileServer)
-				if 0 == clientProfCt && 0 == serverProfCt {
-					sslProf := "Common/clientssl"
-					resources.Virtuals[vKey].AddFrontendSslProfileName(sslProf)
+	for partition, partitionConfig := range resources {
+		for vKey, virtual := range partitionConfig.Virtuals {
+			for _, irule := range virtual.IRules {
+				if strings.Contains(irule, sslPassthroughIRuleName) {
+					clientProfCt := virtual.GetProfileCountByContext(customProfileClient)
+					serverProfCt := virtual.GetProfileCountByContext(customProfileServer)
+					if 0 == clientProfCt && 0 == serverProfCt {
+						sslProf := "Common/clientssl"
+						resources[partition].Virtuals[vKey].AddFrontendSslProfileName(sslProf)
+					}
 				}
 			}
 		}
 	}
+
 	for _, profile := range appMgr.customProfiles.profs {
-		resources.CustomProfiles = append(resources.CustomProfiles, profile)
+		initPartitionData(resources, profile.Partition)
+		resources[profile.Partition].CustomProfiles = append(resources[profile.Partition].CustomProfiles, profile)
 	}
 	for _, irule := range appMgr.irulesMap {
-		resources.IRules = append(resources.IRules, *irule)
+		initPartitionData(resources, irule.Partition)
+		resources[irule.Partition].IRules = append(resources[irule.Partition].IRules, *irule)
 	}
 	for _, intDg := range appMgr.intDgMap {
-		resources.InternalDataGroups = append(resources.InternalDataGroups, *intDg)
+		initPartitionData(resources, intDg.Partition)
+		resources[intDg.Partition].InternalDataGroups = append(resources[intDg.Partition].InternalDataGroups, *intDg)
+	}
+
+	// Update resources to conform to the CCCL schema and empty out unneeded fields
+	// so they will be stripped out by the JSON marshaller
+	for partition, partitionConfig := range resources {
+		for i, _ := range partitionConfig.Virtuals {
```

Comment:
This looks like a good candidate for a sync.WorkGroup and goroutine per for loop.
```
var wg sync.WorkGroup
wg.Add(1)
go runForLoop {
  ...
  wg.Done()
}
wg.Add(1)
go runForLoop {
  ...
  wg.Done()
}
wg.Wait()
```

Reply:
Are you thinking a goroutine per partition or per resource slice (i.e. Virtuals, Pools, etc)?

Reply:
I was thinking more for resource slice but if all these loop iterations work on independent chunks of data then you could go further and make a goroutine per partition that then also creates goroutines to independently work on each of their sections.

Reply:
Agree